core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel_Common.hpp

```diff
@@ -84,9 +84,9 @@ struct ParallelReduceSpecialize<FunctorType, Kokkos::RangePolicy<PolicyArgs...>,
   static void execute_reducer(const FunctorType& f, const PolicyType& p,
                               PointerType result_ptr, bool ptr_on_device) {
     OpenMPTargetExec::verify_is_process(
-        "Kokkos::Experimental::OpenMPTarget parallel_for");
+        "Kokkos::Experimental::OpenMPTarget parallel_reduce");
     OpenMPTargetExec::verify_initialized(
-        "Kokkos::Experimental::OpenMPTarget parallel_for");
+        "Kokkos::Experimental::OpenMPTarget parallel_reduce");
```

Comment:
Wouldn't it be better to call these functions in `execute` directly before the dispatch to `execute_init_join`, `execute_reducer`,...? In that case, you could also be more precise about the policy used in the error message.

Reply:
Edited the error message.